esp-bootloader-esp-idf/src/partitions.rs

```diff
@@ -493,11 +491,19 @@ impl<F> FlashRegion<'_, F> {
     pub fn partition_size(&self) -> usize {
         self.raw.len() as _
     }
+
+    fn range(&self) -> core::ops::RangeInclusive<u32> {
+        self.raw.offset()..=self.raw.offset() + self.raw.len()
```

Comment:
```suggestion
    fn range(&self) -> core::ops::Range<u32> {
        self.raw.offset()..self.raw.offset() + self.raw.len()
```
I think `self.raw.offset() + self.raw.len()` is not part of the partition.